source/simulator/solver_schemes.cc

```diff
@@ -778,11 +795,34 @@ namespace aspect
 
     double relative_residual = std::numeric_limits<double>::max();
     nonlinear_iteration = 0;
+
+    // Store the current location of the particles, so that we can reset
+    // the particles to it in each nonlinear iteration.
+    unsigned int n_locally_owned_particles = 0;
+    std::vector< Point<dim> > local_particle_positions(n_locally_owned_particles);
```

Comment:
since `n_locally_owned_particles` is 0 here there is no need to put it into the constructor. `std::vector< Point<dim> > local_particle_positions;` is always the same as the above.